### ISSUE TITLE
accessibility keyboard support for chart legends

### ISSUE THREAD

**Issue by christian-smith** [apache/echarts#14706]
### What problem does this feature solve?
For accessibility the chart legend labels should be keyboard accessible with a tabindex and should be clickable with the enter key.  Also, the legend prev/next arrow buttons should have the same keyboard controls.

### What does the proposed API look like?
adding tabindex to legend labels and buttons
adding keydown.enter events to legend labels and buttons

**Comment by marmaak** [comment]
Hi!

I know that this is very specific to chart's legend, but since it is a new feature (based on the tag added by @plainheart) can I suggest to take a look at how Highcharts implemented the keyboard navigation for the whole chart? Maybe this issue might be more generic...

https://www.highcharts.com/accessibility/#exploreFeatures

In my company we are evaluating to change the chart library we are using, and se were really impressed by Echarts and all the possibilities it offers, also the documentation and samples are very clear an easy to understand, but one of the most important thing for us is the accessibility and it will be really important to understand which are the plans of Echarts to take the right decision 😃 🙇🏻

**Comment by RFMV34** [comment]
We also need at least keyboard accessibility at least for legends, but ideally for the whole chart.

**Comment by helgasoft** [comment]
[Demo Code](https://codepen.io/helgasoft/pen/YzOymwZ) as proof-of-concept - integration of ECharts and [chart2music](https://github.com/julianna-langston/chart2music).
**Legend keyboard accessibility** - try key '**s**' for scatter and '**b**' for bar chart. 
There is also individual data points highlighting with music(!) and more, press **H** to see hotkey Help.

![image](https://user-images.githubusercontent.com/13038071/220045482-269407dd-9e04-4758-bb98-3fd1d6c05ed5.png)

**Comment by mermonkey** [comment]
@plainheart @Ovilia This is a critical feature. Are we not getting it in 6.0?

**Comment by d-t-w** [comment]
+1 for critical feature!

**Comment by mermonkey** [comment]
Is keyboard accessibility being tracked with any open issue? This is a big gap when comparing to highcharts...

**Comment by helgasoft** [comment]
keyboard accessibility is there when the chart has focus - [Demo](https://echarts.apache.org/examples/en/editor.html?helgasoft=keydown+keyboard&code=PTAEGMHsDsBcCcCWAjArrRNTIJ6gBYCmANgOYCGAzpAGawB0UAtgLABQkADhlgLwDe7UKAyxihAFyh-lVMliEAHrAkByTvEKVKoAILx4kAO4AZQnVCR4eg8YBKiUvlgjIoQk074qiAF6FscnhKVQBfABohV0hiDE4pfgioxV1FREoEkRxOSVAAInByBVIrHDzQJLZhHFT0zIrIqtAAEyKqQhVpKOFqVHhwXIBtbuFQQYAGcOEARnGAXUbR0cHpqdAAJgBWBZHhQfW1gGZtxaWxw7WAFhPdscu19YAOHabhOajKnsIkLSlh166AOEsGyuTyyCCeVOSw8Xh8GWkiAUTAAyiDxAkoMQrGpNM0wp9RqEoi9QgBudjsRDNRSgXgbClsSgdAAqiCYhEg6AAFDRUNBwDxoNyAJSA4RMHAAYW88AYzXSnCK4HwukFmGF_CyOSkqnwjnwxANsFUU1asHIAEloM0lFJqYopsyfpRrbbFBJxhURYyIqBpgGfewjIgbcZ6ORms0AKIAN0IcBM6QU0G-3NUAGtCDhmsZoKbQHyBULQNzOJwxYImogaKXy4xILa6fTVPpDKZzCaxQ66aAALJFfD0JjkRTch0AWlWoHGQertbLnAbTd4Lbb9mNqm7NN7A9gQ6YofHNIA1NOuEL6Ob2gxev1CPRxNBSPup3OJdLZfLFcrVeqYNyWogjqoCqLmRjQJwxDkDgBbOogWhunaM7eoyH4ykE36UEqsAqmqQqAdquR6gaRpOCaZptEhHqgA6TrfAhro2naXqhHObGMuwAAk3K5uAqAcnAIr0JokY4Ly_L_tApaVlE4CxlQvY8XkADEKqYROUBwOQobfHkIoTHMaEQAplD0BayDUcI9IBtMxnyVQ9A0JA_GUKKZJLCAEBfqAqaEM0hYuagOiwG4TAwEiVigFmOZ5u48ZwOwbFAA)
